test/e2e/simple/e2e_test.go

```diff
@@ -0,0 +1,248 @@
+		},
+		{
+			name:    "failure on one failed target",
+			args:    []string{"--target=failure"},
+			success: false,
+			output:  []string{"Container test in pod failure failed, exit code 1, reason Error"},
+		},
+	}
+
+	for _, tc := range testCases {
+		testCase := tc
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			testDone := make(chan struct{})
+			cleanupDone := make(chan struct{})
+			defer func() {
+				close(testDone)
+				<-cleanupDone
+			}()
+			cmd, _ := framework.CiOperatorCommand(t)
+			cmd.Args = append(cmd.Args, append(testCase.args, "--config=config.yaml")...)
+			cmd.Env = append(cmd.Env, `JOB_SPEC={"type":"postsubmit","job":"branch-ci-openshift-ci-tools-master-ci-operator-e2e","buildid":"0","prowjobid":"uuid","refs":{"org":"openshift","repo":"ci-tools","base_ref":"master","base_sha":"6d231cc37652e85e0f0e25c21088b73d644d89ad","pulls":[]}}`)
+			output, err := framework.Execute(t, cmd, testDone, cleanupDone)
+			if testCase.success != (err == nil) {
+				t.Fatalf("%s: didn't expect an error from ci-operator: %v; output:\n%v", testCase.name, err, string(output))
+			}
+			for _, line := range testCase.output {
+				if !bytes.Contains(output, []byte(line)) {
+					t.Errorf("%s: could not find line %q in output; output:\n%v", testCase.name, line, string(output))
+				}
+			}
+		})
+	}
+}
+
+func TestTemplate(t *testing.T) {
+	clusterProfileDir := filepath.Join(t.TempDir(), "cluster-profile")
+	if err := os.MkdirAll(clusterProfileDir, 0755); err != nil {
+		t.Fatalf("failed to create dummy secret dir: %v", err)
+	}
+	if err := ioutil.WriteFile(filepath.Join(clusterProfileDir, "data"), []byte("nothing"), 0644); err != nil {
+		t.Fatalf("failed to create dummy secret data: %v", err)
+	}
+	testDone, cleanupDone := make(chan struct{}), make(chan struct{})
+	defer func() {
+		close(testDone)
+		<-cleanupDone
+	}()
+	cmd, artifactDir := framework.CiOperatorCommand(t)
+	cmd.Args = append(cmd.Args,
+		"--template=template.yaml",
+		"--target=template",
+		"--config=template-config.yaml",
+		"--secret-dir="+clusterProfileDir,
+	)
+	cmd.Env = append(cmd.Env,
+		`CLUSTER_TYPE=something`,
+		`TEST_COMMAND=executable`,
+		`JOB_SPEC={"type":"postsubmit","job":"branch-ci-openshift-ci-tools-master-ci-operator-e2e","buildid":"0","prowjobid":"uuid","refs":{"org":"openshift","repo":"ci-tools","base_ref":"master","base_sha":"6d231cc37652e85e0f0e25c21088b73d644d89ad","pulls":[]}}`,
+	)
+	output, err := framework.Execute(t, cmd, testDone, cleanupDone)
+	if err != nil {
+		t.Fatalf("ci-operator failed: %v; output:\n%v", err, string(output))
+	}
+	framework.CompareWithFixtureDir(t, "artifacts/template", filepath.Join(artifactDir, "template"))
+	outputjUnit := filepath.Join(artifactDir, "junit_operator.xml")
+	raw, err := ioutil.ReadFile(outputjUnit)
+	if err != nil {
+		t.Fatalf("could not read jUnit artifact: %v", err)
+	}
+	matcher := regexp.MustCompile(`time=".*"`)
```

Comment:
compile regexes to a var please so this fails during startup if there is an issue with it